public/login.js

```diff
@@ -1,26 +1,79 @@
-function login(e) {
-  e.target.action = `https://api.core.ac.uk/login_check?continue=${encodeURIComponent(
-    window.location.origin
-  )}`
-  window.dispatchEvent(new Event('login-in-processing'))
-}
-
 function addLoadingAnimation() {
   const form = document.getElementById('login-form')
   form.classList.add('loading')
 }
 
+function removeLoadingAnimation() {
+  const form = document.getElementById('login-form')
+  form.classList.remove('loading')
+}
+
+function handlePostMessage(event) {
+  if (event.data === 'login-finished') removeLoadingAnimation()
+}
+
+function showWrongCredentialsMessage() {
+  const message = document.getElementById('message')
+  message.innerHTML = 'The username or password you entered is incorrect.'
+  message.classList.add('message-error')
+  message.classList.remove('message-info')
+}
+
+function showLogoutMessage() {
+  const message = document.getElementById('message')
+  message.innerHTML = 'You have been successfully logged out. Log in again.'
+  message.classList.add('message-info')
+  message.classList.remove('message-error')
+}
+
+function login(event) {
+  event.preventDefault()
+  const urlParams = new URLSearchParams(window.location.search)
+  const identityProviderUrl =
+    urlParams.get('identity_provider_url') || 'https://api.core.ac.uk/'
+
+  window.dispatchEvent(new Event('login-processing'))
+  const formData = new FormData(event.target)
+  const data = new URLSearchParams(formData)
+  fetch(`${identityProviderUrl}login_check`, {
+    method: 'POST',
+    body: data,
+    credentials: 'include',
+  })
+    .then(r => {
+      if (r.status === 401) showWrongCredentialsMessage()
+    })
+    .finally(() => {
+      if (window.location !== window.parent.location) {
+        // The page is in an iframe
+        window.parent.postMessage('login-processing', window.location.origin)
+      } else window.location = '/'
+    })
+
+  return false
+}
+
 window.addEventListener('DOMContentLoaded', () => {
   const urlParams = new URLSearchParams(window.location.search)
-  if (urlParams.has('reason')) {
-    const message = document.getElementById('error-message')
-    message.innerHTML = 'The username or password you entered is incorrect.'
+  const reason = urlParams.get('reason')
+  if (reason) {
+    switch (reason) {
+      case 'wrong_credentials':
+        showWrongCredentialsMessage()
+        break
+      case 'logout':
+        showLogoutMessage()
+        break
+      default:
+    }
   }
 
-  if (urlParams.has('loading')) addLoadingAnimation()
+  if (urlParams.get('loading') === 'true') addLoadingAnimation()
```

Comment:
It looks weird, prefer checking for existence of the parameter and pass it properly from the parent.

Reply:
https://github.com/oacore/dashboard/pull/186/commits/1224a9bd1e665960160b0b6de07c3ad028a2833d